Lib/pdb.py

```diff
@@ -304,6 +304,14 @@ def setup(self, f, tb):
         # cache it here to ensure that modifications are not overwritten.
         self.curframe_locals = self.curframe.f_locals
         self.set_convenience_variable(self.curframe, '_frame', self.curframe)
+
+        if self._chained_exceptions:
+            self.set_convenience_variable(
+                self.curframe,
+                '_exception',
+                self._chained_exceptions[self._chained_exception_index],
```

Comment:
I don't remember- are we sure here that the index is valid?

Reply:
First of all, yes, we only update `self._chained_exception_index` when there's a traceback with it. The thing is, we actuall consider the exception "invalid" if there's no traceback attached to it - but the exception is still there. So even in this circumstance, the exception will exist and there will be no error.